docs/guides/advanced-routing.md

```diff
@@ -0,0 +1,332 @@
+# Advanced HTTP Routing
+
+In this guide we will be configuring a more advanced set of routing rules with multiple applications. These rules
+will showcase different paths, parameters, and methods used to fine-tune the flow of traffic. For an introduction
+to exposing your application, it is recommended to go through the
+[basic guide](/docs/guides/how-to-direct-traffic-to-your-app.md) first.
+
+The following image shows the traffic flow that we will be creating with these rules.
+
+![Traffic Flow Diagram](/docs/images/advanced-routing.png)
+
+The goal is to create a set of rules that will result in users' requests being sent to specific backends based on
+the conditions in which the request is sent.
+
+## Prerequisites
+
+- NGINX Kubernetes Gateway is [installed](/docs/installation.md).
+- [Expose NGINX Kubernetes Gateway](/docs/installation.md#expose-nginx-kubernetes-gateway) and save the public IP
+  address and port of NGINX Kubernetes Gateway into shell variables:
+
+   ```text
+   GW_IP=XXX.YYY.ZZZ.III
+   GW_PORT=<port number>
+   ```
+
+> In a production environment, you should have a DNS record for the external IP address that is exposed,
+> and it should refer to the hostname that the gateway will forward for.
+
+## Coffee applications
+
+### Deploy the coffee applications
+
+Begin by deploying the `coffee-v1` and `coffee-v2` applications:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-kubernetes-gateway/main/examples/advanced-routing/coffee.yaml
```

Comment:
I think we should get these examples to be stable enough where we don't need to do that.

Reply:
I think it is a good goal. But, considering that our documentation is versioned (via tags), it could be a case that an example changes and doesn't match the contents of the guide.